socs/agents/galildmc_axis_controller/agent.py

```diff
@@ -0,0 +1,1170 @@
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            self.stage.enable_sin_commutation(axis)
+            self.log.info(f'Sin commutation setting for {axis} is enabled.')
+
+        return True
+
+    @ocs_agent.param('axis', type=str)
+    def disable_limit_switch(self, session, params):
+        """disable_limit_switch(axis)
+
+        **Task** - Disable the hardware limit switch for a specified axis.
+
+        Parameters:
+            axis (str): Specified axis. Ex. 'A'
+
+        """
+        axis = params['axis']
+        with self.lock.acquire_timeout(timeout=5, job='disable_limit_switch') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            self.stage.disable_limit_switch(axis)
+            self.log.info(f'Limit switch for {axis} is disabled.')
+
+        return True, f'Limit switch for {axis} is disabled.'
+
+    @ocs_agent.param('polarity', type=int)
+    def set_limitswitch_polarity(self, session, params):
+        """set_limitswitch_polarity(polarity)
+
+        **Task** - Configure the limit switch polarity for all axes.
+
+        Parameters:
+            polarity (int): Limit switch polarity value.
+                0 is active-low (limit triggers when input goes low),
+                1 is active-high (limit triggers when input goes high).
+
+        """
+        pol = params['polarity']
+        with self.lock.acquire_timeout(timeout=5, job='set_limitswitch_polarity') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            self.stage.set_limitswitch_polarity(pol)
+            self.log.info(f'Limit switch polarity for {axis} is set to {pol}.')
+
+        return True
+
+    @ocs_agent.param('axis', type=str)
+    def stop_axis_motion(self, session, params):
+        """stop_axis_motion()
+
+        **Task** - Stop motion for specified axis.
+
+        Parameters:
+              axis (str): Specified axis. Ex. 'A'
+
+        """
+        axis = params['axis']
+        with self.lock.acquire_timeout(timeout=5, job='stop_axis_motion') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            self.stage.stop_motion(axis)
+
+        return True, f'Stopped motion for {axis}.'
+
+    @ocs_agent.param('order', type=str)
+    def set_gearing(self, session, params):
```

Comment:
Rename this (and its related driver function) to give more context and differentiate from `set_gearing_ratio`.